File: fibonacci_number/FibonacciNumber.java
public class FibonacciNumber {

    // fibonacci function return the nth fibonacci no for all n>0
    public static int fibonacci(int n) {
        if (n == 1 || n == 2) {
            return (n - 1);
        } else {
            // Store second last fibonacci number
            int a = 0;
            // Store last fibonacci number
            int b = 1;
            // Store current fibonacci number which is sum of last and second last fibonacci no
            int fib = 0;
            for (int i = 2; i < n; i++) {
                fib = a + b;
                a = b;
                b = fib;
            }
            return (fib);
        }
    }

    public static void main(String[] args) {
        // n>0
        int n = 5;
        if (n <= 0) {
            System.out.println("n must be greater than 0");
            return;
        }
        System.out.println(fibonacci(n));
    }
}

File: fibonacci_number/fibonacciNumber.c
// C program to find the nth fibonacci number.
// @author Manas Gupta (@heisenberg-2505)
#include <stdio.h>
typedef long long LL;

// Function to calculate the nth fibonacci number.
LL fibonacci (LL n) {
  if (n == 1 || n == 2) {
    return (n - 1);
  } else {
    // Variable to store the second last fibonacci number wrt current fibonacci number.
    LL secondLastFib = 0;
    // Variable to store the last fibonacci number wrt current fibonacci number.
    LL lastFib = 1;
    // Variable to store the current fibonacci number.
    LL currentFib = 0;
    for (LL i = 2; i < n; i++) {
      currentFib = secondLastFib + lastFib;
      secondLastFib = lastFib;
      lastFib = currentFib;
    }
    return currentFib;
  }
}

int main () {
  // Change here to change the value of n.
  LL n = 10;
  if (n <= 0) {
    printf ("The value of n should be a positive integer.\n");
    return 1;
  }
  printf ("The %lldth fibonacci number is %lld \n", n, fibonacci (n));
  return 0;
}
